packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -436,23 +437,23 @@ class _HelperErrorState extends State<_HelperError> with SingleTickerProviderSta
 
     if (_controller.isCompleted) {
       _helper = null;
-      if (widget.errorText != null) {
+      if (widget.errorText != null || widget.error != null) {
```

Comment:
Nit: It might be worth creating a getter for this since it's used pretty frequently.

```dart
bool get _hasError => widget.errorText != null || widget.error != null;
```